webgpu.h

```diff
@@ -575,6 +574,15 @@ typedef enum WGPUVertexStepMode {
     WGPUVertexStepMode_Force32 = 0x7FFFFFFF
 } WGPUVertexStepMode;
 
+typedef enum WGPUSurfaceStatus {
```

Comment:
Replaced `bool * suboptimal` with an enum.
This matches [`SurfaceStatus`](https://docs.rs/wgpu-types/latest/wgpu_types/enum.SurfaceStatus.html) in wgpu.

But still not sure if we should only have `Good` & `Suboptimal` and report others as errors.

Reply:
I'm not sure what Outdated means in practice, but the enum sounds good.

Reply:
>I'm not sure what Outdated means in practice

in wgpu it is [mapped](https://github.com/gfx-rs/wgpu/blob/f7526ae9604a91d14e58eea0d6342988b4a94e79/wgpu-hal/src/vulkan/instance.rs#L744) to `VK_ERROR_OUT_OF_DATE_KHR`

after thinking more about it, WGPUSurfaceStatus can be
```c
typedef enum WGPUSurfaceStatus {
    WGPUSurfaceStatus_Good = 0x00000000,  // ~ VK_SUCCESS
    WGPUSurfaceStatus_Suboptimal = 0x00000001, // ~ VK_SUBOPTIMAL_KHR
    WGPUSurfaceStatus_Timeout = 0x00000002, // ~ VK_TIMEOUT
    WGPUSurfaceStatus_Force32 = 0x7FFFFFFF
} WGPUSurfaceStatus;
```
removed Lost & Outdated, those can be reported as errors.